backend/src/main/java/de/bund/digitalservice/ris/caselaw/domain/DocumentationUnitService.java

```diff
@@ -447,30 +448,29 @@ public RisJsonPatch updateDocumentationUnit(
 
       if (!toUpdate.getOperations().isEmpty()) {
         toUpdate = patchMapperService.removeTextCheckTags(toUpdate);
+        patchedDocumentationUnitWithBase64Images =
+            patchMapperService.applyPatchToEntity(toUpdate, existingDocumentationUnit);
+        toUpdate =
+            patchMapperService.extractAndStoreBase64Images(
+                toUpdate, existingDocumentationUnit, null);
 
         DocumentationUnit patchedDocumentationUnit =
             patchMapperService.applyPatchToEntity(toUpdate, existingDocumentationUnit);
 
-        if (patchedDocumentationUnit instanceof Decision docUnit) {
-          patchedDocumentationUnit = docUnit.toBuilder().version(newVersion).build();
-        } else if (patchedDocumentationUnit instanceof PendingProceeding pendingProceeding) {
-          patchedDocumentationUnit = pendingProceeding.toBuilder().version(newVersion).build();
-        }
+        patchedDocumentationUnit =
+            updateDocumentationUnitVersion(patchedDocumentationUnit, newVersion);
+
+        patchedDocumentationUnitWithBase64Images =
+            updateDocumentationUnitVersion(patchedDocumentationUnitWithBase64Images, newVersion);
```

Comment:
Can't we set the new version on `existingDocumentationUnit` in Line 450 already?